test/hawkes_test.jl

```diff
@@ -112,57 +109,97 @@ uselrate[3] = true
-        sols[n] = solve(jump_prob, stepper)
-    end
-    if alg isa Coevolve
-        λs = permutedims(mapreduce((sol) -> empirical_rate(sol), hcat, sols))
-    else
-        cols = length(sols[1].u[1].u)
-        λs = permutedims(mapreduce((sol) -> empirical_rate(sol), hcat, sols))[:, 1:cols]
+    for vr_aggregator in (VR_FRM(), VR_Direct())
+        if alg isa Coevolve
+            stepper = SSAStepper()
+        else
+            stepper = Tsit5()
+        end
+        sols = Vector{ODESolution}(undef, Nsims)
+        for n in 1:Nsims
+            jump_prob = hawkes_problem(p, alg; u = u0, tspan, g, h, uselrate = uselrate[1], vr_aggregator = vr_aggregator)
+
+            reset_history!(h)
+            if stepper == Tsit5()
+                sols[n] = solve(jump_prob, stepper)
+            else
+                sols[n] = solve(jump_prob, stepper)
+            end
+        end
+
+        if alg isa Coevolve
+            λs = permutedims(mapreduce((sol) -> empirical_rate(sol), hcat, sols))
+        else
+            if vr_aggregator isa VR_FRM
+                cols = length(sols[1].u[1].u)
+
+                λs = permutedims(mapreduce((sol) -> empirical_rate(sol), hcat, sols))[:, 1:cols]
+
+                @test isapprox(mean(λs), Eλ; atol = 0.01)
+                @test isapprox(var(λs), Varλ; atol = 0.001)
+            else
+                cols = length(sols[1].u[1])
+
+                λs = permutedims(mapreduce((sol) -> empirical_rate(sol), hcat, sols))
+
+                @test isapprox(mean(λs), Eλ; atol = 0.01)
+                @test isapprox(var(λs), Varλ; atol = 0.001)
+            end
+        end
     end
-    @test isapprox(mean(λs), Eλ; atol = 0.01)
-    @test isapprox(var(λs), Varλ; atol = 0.001)
 end
 
 # test stepping Coevolve with continuous integrator and bounded jumps
 let alg = Coevolve()
-    oprob = ODEProblem(f!, u0, tspan, p)
-    jumps = hawkes_jump(u0, g, h)
-    jprob = JumpProblem(oprob, alg, jumps...; dep_graph = g, rng)
-    @test ((jprob.variable_jumps === nothing) || isempty(jprob.variable_jumps))
-    sols = Vector{ODESolution}(undef, Nsims)
-    for n in 1:Nsims
-        reset_history!(h)
-        sols[n] = solve(jprob, Tsit5())
+    for vr_aggregator in (VR_FRM(), VR_Direct())
+        oprob = ODEProblem(f!, u0, tspan, p)
+        jumps = hawkes_jump(u0, g, h)
+        jprob = JumpProblem(oprob, alg, jumps...; vr_aggregator = vr_aggregator, dep_graph = g, rng)
+        @test ((jprob.variable_jumps === nothing) || isempty(jprob.variable_jumps))
+        sols = Vector{ODESolution}(undef, Nsims)
+        for n in 1:Nsims
+            jprob = JumpProblem(oprob, alg, jumps...; vr_aggregator = vr_aggregator, dep_graph = g, rng)
+
+            reset_history!(h)
+            sols[n] = solve(jprob, Tsit5())
+        end
+        λs = permutedims(mapreduce((sol) -> empirical_rate(sol), hcat, sols))
+        @test isapprox(mean(λs), Eλ; atol = 0.01)
+        @test isapprox(var(λs), Varλ; atol = 0.001)
```

Comment:
This needs to be reverted, and modified as minimally as possible. It should just require creating an outer loop over the two variable rate integrators and not all these other changes. Again, you shouldn't be creating a new jump problem for every call to solve.